subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/swift/CachingCompilerMetaDataProvider.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.toolchain.internal.swift;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CachingCompilerMetaDataProvider implements CompilerMetaDataProvider {
+
+    private final CompilerMetaDataProvider delegate;
+    private final Map<File, SwiftcVersionResult> resultMap = new HashMap<File, SwiftcVersionResult>();
+
+    public CachingCompilerMetaDataProvider(CompilerMetaDataProvider delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public SwiftcVersionResult getSwiftcMetaData(File swiftc) {
```

Comment:
Probably makes sense to have a unit test to verify that it actually reuses a cached result.

Reply:
Added.